### PR TITLE
Add NUMA topology support

### PR DESCRIPTION
**Pull request by stephenfin**
I'm not a ruby developer so apologies if I've done something daft. Suggestions of how I can test this would be appreciated.

### CODE REVIEW

**Comment by infernix** [comment]
Think there's an error in your _generate_numa since Travis build fails; can you try to fix it?

Also, please remove the tab fix commit in non-related code; happy to take cosmetic fixes but shouldn't be in a PR that adds functionality :)

File: lib/vagrant-libvirt/action/create_domain.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def initialize(app, _env)
         end
 
         def _disk_name(name, disk)
-          "#{name}-#{disk[:device]}.#{disk[:type]}"	# disk name
+          "#{name}-#{disk[:device]}.#{disk[:type]}"  # disk name
```
Comment (pronix) [inline lib/vagrant-libvirt/action/create_domain.rb]: this change is not required.

File: lib/vagrant-libvirt/config.rb [hunk 2]
```diff
@@ -198,15 +200,15 @@ def initialize
       end
 
       def boot(device)
-        @boot_order << device	# append
+        @boot_order << device  # append
```
Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: this, 209 211 - changes for nothing.

File: lib/vagrant-libvirt/config.rb [hunk 3]
```diff
@@ -457,6 +482,8 @@ def finalize!
         @cpu_model = 'qemu64' if @cpu_model == UNSET_VALUE
         @cpu_fallback = 'allow' if @cpu_fallback == UNSET_VALUE
         @cpu_features = [] if @cpu_features == UNSET_VALUE
+        @numa_nodes = nil if @numa_nodes == UNSET_VALUE
```
Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: why require redefine varaiable if value not set ?

**Comment by infernix** [comment]
@stephenfin Does this work for any value of mem, even not cleanly divisable in numa_nodes?

**Comment by stephenfin** [comment]
@infernix No, the memory would also have to be cleanly divisible into numa_nodes. I guess we could either assume everyone's a consenting adult and won't pass in something daft, or we could raise an exception if it didn't divide cleanly. If we opt for the latter, I'd appreciate a little help as, again, not a ruby dev :smile:

**Comment by infernix** [comment]
@stephenfin always assume user isn't bright ;)

**Pull request merged**